core/src/main/java/org/elasticsearch/action/update/UpdateResponse.java

```diff
@@ -44,47 +43,16 @@ public UpdateResponse() {
      * Constructor to be used when a update didn't translate in a write.
      * For example: update script with operation set to none
      */
-    public UpdateResponse(String index, String type, String id, long version, boolean created) {
-        this(new ShardInfo(0, 0), index, type, id, version, created);
+    public UpdateResponse(ShardId shardId, String type, String id, long version, boolean created) {
+        this(new ShardInfo(0, 0), shardId, type, id, -1l, version, created);
```

Comment:
Another `-1l` I'd consider replacing with the field [proposed previously](https://github.com/elastic/elasticsearch/pull/14651/files#r45225330).